components/script/dom/document.rs

```diff
@@ -2154,7 +2153,7 @@ impl Document {
                                         HasBrowsingContext::No,
                                         None,
                                         // https://github.com/whatwg/html/issues/2109
-                                        Origin::opaque_identifier(),
+                                        self.origin.clone(),
```

Comment:
Well the latest opinion on that discussion was @smaug---- saying https://github.com/whatwg/html/issues/2109#issuecomment-263957057 which looks like it's coming down on the side of a template inheriting its origin, rather than generating a new opaque origin. Can the template document cause script to run? If so, we may need to make sure it runs in the right script thread.